lib/matplotlib/patches.py

```diff
@@ -499,11 +499,14 @@ def draw(self, renderer):
 
         gc.set_foreground(self._edgecolor, isRGBA=True)
 
-        lw = self._linewidth
-        if self._edgecolor[3] == 0:
-            lw = 0
-        gc.set_linewidth(lw)
-        gc.set_linestyle(self._linestyle)
+        if (self._edgecolor[3] == 0 or 
+            self._linestyle in {'none', 'None', ' ', '', None}):
```

Comment:
Please do not include `None` in this check.  We reserve `None` to mean 'do the default thing' not 'do nothing'.